### PR TITLE
Git staging support for rename.sh

### PR DESCRIPTION
**Pull request by ilkka**
This PR adds support for transparently git staging the results of running the rename script, if git is available *and* the root of the app template project is a git repo.

It has two parts: the first just refactors the script a bit to get rid of the repetition and to make it possible to make the rename commands go through git. The second adds the actual git support. Tested in mac OS Sierra, seems to otherwise work 100% like the original.

### CODE REVIEW

File: support/rename.sh [hunk 1]
```diff
@@ -7,18 +7,32 @@ appRoot=`dirname $0`/..
 
 [ -z "${newAppName}" ] && echo 'Missing required parameter newAppName' && exit 1
 
-grep -rI 'PepperoniAppTemplate' --exclude='rename.sh' $appRoot/* | tr ':' ' ' | awk '{print $1}' | uniq | xargs -I{} sed -i.bak "s/PepperoniAppTemplate/${newAppName}/g" {}
-grep -rI 'pepperoniapptemplate' --exclude='rename.sh' $appRoot/* | tr ':' ' ' | awk '{print $1}' | uniq | xargs -I{} sed -i.bak "s/pepperoniapptemplate/${newLowerCaseName}/g" {}
-find . -name '*.bak' -exec rm {} \;
+# gather all modification targets
+filesToModify=$(grep -riIl 'pepperoniapptemplate' --exclude='rename.sh' $appRoot/*)
```
Comment (ilkka) [inline support/rename.sh]: Here the POSIX standard `-l` flag to grep does the same thing as the tr+awk+uniq pipeline in the original, and the `-i` flag removes the need for two different patterns.

File: support/rename.sh [hunk 2]
```diff
@@ -7,18 +7,32 @@ appRoot=`dirname $0`/..
...
+  mvCmd="mv"
+fi
+
+for fileToRename in $filesToRename; do
+  newName=$(echo $fileToRename | sed "s/PepperoniAppTemplate/$newAppName/g;s/pepperoniapptemplate/$newLowerCaseName/g")
+  mkdir -p $(dirname "$newName") && $mvCmd "$fileToRename" "$newName"
 done
 
+# remove leftover empty directories
+rmdir -p $(find "$appRoot" -ipath '*pepperoniapptemplate*' -type d) 2>/dev/null
```
Comment (ilkka) [inline support/rename.sh]: `rmdir -p` tries to remove every path component of its arguments, and stops when it encounters non-empty directories. The output redirection swallows the spurious error message.

File: support/rename.sh [hunk 3]
```diff
@@ -7,18 +7,32 @@ appRoot=`dirname $0`/..
...
+for fileToModify in $filesToModify; do
+  sed -i.bak "s/PepperoniAppTemplate/${newAppName}/g;s/pepperoniapptemplate/${newLowerCaseName}/g" $fileToModify
 done
+find "${appRoot}" -name '*.bak' -exec rm {} \;
 
-for fileToMove in `find $appRoot/android -depth -name '*pepperoniapptemplate*'`; do
-  mv $fileToMove `echo $fileToMove | sed "s/\(.*\)pepperoniapptemplate/\1$newLowerCaseName/g"`
+if [ -d "${appRoot}/.git" -a -n "$(command -v git)" ]; then
+  # stage all string replacements and set up to stage renames, below
+  git add --update
```
Comment (chadrien) [inline support/rename.sh]: I think it's ok the way it is. Let see if we have issues regarding this. If we don't, it means we don't need to fix what ain't broke.

**Comment by krivachy** [comment]
Epic stuff, thanks for the contribution! 🎉 🍕 🚀

**Pull request merged**